test/unit/get-page-static-infos.test.ts

```diff
@@ -0,0 +1,37 @@
+/* eslint-env jest */
+import { getMiddlewareMatchers } from 'next/dist/build/analysis/get-page-static-info'
+
+describe('get-page-static-infos', () => {
+  describe('getMiddlewareMatchers', () => {
```

Comment:
Do these tests demonstrate how the `source` wont be modified? Would like to see something like that

Reply:
The `source` value is changed within this function and it's not returned from teh function. I don't think there's much we should do here.

The use of this function is covered in other integration tests, too.